engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/consumerrecord/ConsumerRecordToJsonFormatterFactory.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.engine.kafka.consumerrecord
+
+import java.nio.charset.StandardCharsets
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder}
+import org.apache.flink.streaming.connectors.kafka.KafkaDeserializationSchema
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import pl.touk.nussknacker.engine.api.CirceUtil
+import pl.touk.nussknacker.engine.api.test.{TestDataSplit, TestParsingUtils}
+import pl.touk.nussknacker.engine.kafka.{KafkaConfig, RecordFormatter, RecordFormatterFactory}
+
+import scala.reflect.ClassTag
+
+/**
+  * RecordFormatter used to encode and decode whole raw kafka event (ConsumerRecord) in json format.
+  *
+  * @param deserializationSchema - schema used to convert raw kafka event to serializable representation (see SerializableConsumerRecord)
+  * @param serializeKeyValue - converts key and value from K-V domain to Array[Byte]
+  * @tparam K - event key type
+  * @tparam V - event value type
+  */
+class ConsumerRecordToJsonFormatterFactory[K:Encoder:Decoder, V:Encoder:Decoder] extends RecordFormatterFactory {
+
+  override def create[KK: ClassTag, VV: ClassTag](kafkaConfig: KafkaConfig, kafkaSourceDeserializationSchema: KafkaDeserializationSchema[ConsumerRecord[KK, VV]]): RecordFormatter = {
+    new RecordFormatter {
```

Comment:
Can you extract it to separate class? Let it be in the same file. Currently it is hard to override this class. I know that "prefer composition over inheritance" rule but sometimes... ;)